sycl/doc/design/OffloadDesign.md

```diff
@@ -0,0 +1,282 @@
+responsible for performing any of the link-time processing that occurs for
+each device target.
+
+The updated offloading model removes much of the functionality that is
+performed during the link phase of the offloading compilation and moves it
+to a `clang-linker-wrapper` tool.
+
+Below is a general representation of the overall offloading flow that is
+performed during a full compilation from source to final executable.  The
+compiler driver is responsible for creating the fat object and the
+`clang-linker-wrapper` tool is responsible for the general functionality that is
+performed during the link.
+
+![High level view of the offloading flow](images/OffloadGeneralFlow.png
+ "General Offload Flow")
+
+*Diagram 1: General Offload Flow*
+
+More specific details of the clang-linker-wrapper behavior can be seen in the
+[clang-linker-wrapper](#clang-linker-wrapper) section.
+
+## Fat Binary Representation
+
+Binaries generated during the offload compilation will be 'bundled' together
+to create a conglomerate fat binary.  Depending on the type of binary, the
+packaging of the device section will be different.
+
+## Device only Binaries
+
+Generation and consumption of device only binaries is also available.
+
+## Fat Binary Generation
+
+The generation of the fat binary will take place in the driver.  The model from
+the community is generating the fat binary as a secondary process when creating
+the host object.  Generation will be separated from the host compilation step.
+This is being done to enable proper support for using an external host compiler
+as well as taking advantage of potential parallelism during compilation of both
+the host and target device binaries.
+
+The fat object in the new model is generated during the host compilation.
+The host compilation takes an additional argument which points to the device
+binary which will be embedded in the final object.  Generation will be
+separated out to allow for potential parallelism during compilation of both the
+host and target device binaries.
+
+![Creating the fat object](images/OffloadFatObject.png
+ "Fat Object Generation")
+
+*Diagram 2: Fat Object Generation*
+
+### Packager
+
+When dealing with multiple device binaries, an additional step is performed to
+package the multiple device binaries before being added to the host object.
+This additional step is performed with the `clang-offload-packager` taking
+image inputs containing information relating to the target triple,
+architecture setting and offloading kind.
+
+The `clang-offload-packager` is run during 'fat object' generation regardless
+of the number of device binaries being added to the conglomerate fat object.
+The device binaries are contained in what is designated as an ‘Offload Binary’.
+These binaries can reside in a variety of binary formats including Bitcode
+files, ELF objects, executables and shared objects, COFF objects, archives or
+simply stored as an offload binary.
```

Comment:
SPIR-V doesn't look to be supported.  This would need to be added.  As you mentioned below, linking of native/SPIR-V is now possible having the ability to create a SPIR-V based fat object is desired.

Reply:
Effectively, it looks like it would just be the SPIR-V device binary embedded in whatever the host object is (device, object, etc) in their respective sections, of course packaged before added to the section.